internal/persistence/ent/client/batch.go

```diff
@@ -0,0 +1,31 @@
+package entclient
+
+import "iter"
+
+var defaultBatchSize = 100
+
+// batch groups elements from the provided sequence into batches of size
+// defaultBatchSize, yielding a stream of slices, each containing at most
+// defaultBatchSize elements, except for the last batch which may contain fewer.
+// Note that the slices are reused internally to minimize allocations, so they
+// should not be retained or modified after being yielded to avoid unexpected
+// behavior.
+func batch[T any](seq iter.Seq[T]) iter.Seq[[]T] {
```

Comment:
It doesn't seem like there is any code specific to ent in the batch function — I think it would be nice to make it a separate batch package that could be used other places in Enduro (or other projects).  The only change I would make it more reusable as package would be the ability to override the default batch size when calling the function.

Reply:
One option could be `github.com/artefactual-labs/gotools/iterutil`, what do you think?